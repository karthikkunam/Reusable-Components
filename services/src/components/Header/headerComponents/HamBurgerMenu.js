import React, { useState } from 'react';
import PropTypes from 'prop-types';

const HamBurgerMenu = props => {
  const [selectedIndex, setSelectedIndex] = useState(props.defaultLeftNav);
  const [selectedMenu, setSelectedMenu] = useState('');
  const { leftNavItems, expanded } = props;

  const onLeftMenuClick = (selectedMenu, subMenus, index) => {
    const { handleLeftMenu } = props;
    handleLeftMenu && handleLeftMenu(subMenus);
    setSelectedIndex(index);
    setSelectedMenu(selectedMenu);
  };
  return (
    <div className="body">
      <ul className="list">
        {leftNavItems &&
          leftNavItems.length > 0 &&
          leftNavItems.map((item, index) => {
            return (
              <li key={item.leftMenu} className="list-item">
                <a
                  role="button"
                  data-toggle="collapse"
                  data-target="#navbarNavDropdown11"
                  aria-controls="navbarNavDropdown11"
                  aria-expanded={expanded}
                  aria-label={`Toggle navigation ${selectedMenu}`}
                  className={`list-item-link ${index === selectedIndex ? 'active' : ''}`}
                  onClick={() => {
                    onLeftMenuClick(item.leftMenu, item.subMenus, index);
                  }}>
                  {item.leftMenu}
                </a>
              </li>
            );
          })}
      </ul>
      <div className="footer">
        <ul className="list logout">
          <li className="list-item">
            <a className="list-item-link">
              <label>logout</label>
            </a>
          </li>
        </ul>
        <div className="version-text">
          <span>Version Information</span>
        </div>
      </div>
    </div>
  );
};
HamBurgerMenu.defaultProps = {
  defaultLeftNav: null,
};

HamBurgerMenu.propTypes = {
  leftNavItems: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string,
      iconClass: PropTypes.string,
      url: PropTypes.string,
    }),
  ),
  defaultLeftNav: PropTypes.number,
  handleLeftMenu: PropTypes.func,
  expanded: PropTypes.bool,
};

export default HamBurgerMenu;
