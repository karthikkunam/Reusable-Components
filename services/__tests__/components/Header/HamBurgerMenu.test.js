import React from 'react';
import HamBurgerMenu from '../../../src/components/Header/headerComponents/HamBurgerMenu';
import { shallow } from 'enzyme';

// const onNavigationChange = jest.fn();
const props = {
  leftNavItems: [
    {
      leftMenu: 'home',
      subMenus: [
        {
          header: 'Header 1',
          subMenuList: [
            {
              title: 'page 1 Title',
              url: 'page1',
            },
          ],
        },
        {
          header: 'Header 2',
          subMenuList: [
            {
              title: 'page 4 Title',
              url: 'page4',
            },
          ],
        },
      ],
    },
  ],
  handleLeftMenu: jest.fn(),
};

describe('HamBurgerMenu Component', () => {
  it('renders correctly', () => {
    const component = shallow(<HamBurgerMenu {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders onLeftMenuClick onClick', () => {
    const component = shallow(<HamBurgerMenu {...props} />);
    expect(component).toMatchSnapshot();
    component
      .find('li')
      .at(0)
      .find('a')
      .simulate('click', {
        onLeftMenuClick: () => {},
      });
  });
});
