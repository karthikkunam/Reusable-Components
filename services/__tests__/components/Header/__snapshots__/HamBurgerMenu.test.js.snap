// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamBurgerMenu Component renders correctly 1`] = `
<div
  className="body"
>
  <ul
    className="list"
  >
    <li
      className="list-item"
      key="home"
    >
      <a
        className="list-item-link menu-open "
        onClick={[Function]}
      >
        <label>
          home
        </label>
      </a>
    </li>
  </ul>
  <div
    className="footer"
  >
    <ul
      className="list logout"
    >
      <li
        className="list-item"
      >
        <a
          className="list-item-link"
        >
          <label>
            logout
          </label>
        </a>
      </li>
    </ul>
    <div
      className="version-text"
    >
      <span>
        Version Information
      </span>
    </div>
  </div>
</div>
`;

exports[`HamBurgerMenu Component renders onLeftMenuClick onClick 1`] = `
<div
  className="body"
>
  <ul
    className="list"
  >
    <li
      className="list-item"
      key="home"
    >
      <a
        className="list-item-link menu-open "
        onClick={[Function]}
      >
        <label>
          home
        </label>
      </a>
    </li>
  </ul>
  <div
    className="footer"
  >
    <ul
      className="list logout"
    >
      <li
        className="list-item"
      >
        <a
          className="list-item-link"
        >
          <label>
            logout
          </label>
        </a>
      </li>
    </ul>
    <div
      className="version-text"
    >
      <span>
        Version Information
      </span>
    </div>
  </div>
</div>
`;
